awex-core/src/main/java/com/raycoarana/awex/Awex.java

```diff
@@ -114,6 +120,61 @@ void submit(Runnable runnable) {
         }
     }
 
+	/**
+     * Creates a new promise that will be resolved with the result of the first task which completes correctly.
+     * Task will execute sequentially until the first one is resolved.
+     * If all the promises fail promise will be rejected with the last exception thrown.
+     * In case the collection of tasks is empty the promise will be rejected with an EmptyTasksException.
+     *
+     * @param tasks source tasks to execute sequentially
+     * @param <Result> type of result of the promises
+     * @param <Progress> type of progress of the promises
+     * @returna new promise that will be resolved with the first task completed correctly
+     */
+    @SafeVarargs
+    public final <Result, Progress> Promise<Result, Progress> runUntilFirstDone(final
+    Task<Result, Progress>... tasks) {
+        return runUntilFirstDone(Arrays.asList(tasks));
+    }
+
+    /**
+     * Creates a new promise that will be resolved with the result of the first task which completes correctly.
+     * Task will execute sequentially until the first one is resolved.
+     * If all the promises fail promise will be rejected with the last exception thrown.
+     * In case the collection of tasks is empty the promise will be rejected with an EmptyTasksException.
+     *
+     * @param tasks source tasks to execute sequentially
+     * @param <Result> type of result of the promises
+     * @param <Progress> type of progress of the promises
+     * @returna new promise that will be resolved with the first task completed correctly
+     */
+    public <Result, Progress> Promise<Result, Progress> runUntilFirstDone(final
```

Comment:
To not be confused with anyOf() operation, I would name it more explicitly about it's sequential. Maybe sequentiallyUntilFirstDone().